### PR TITLE
handling 402

### PR DESCRIPTION
**Pull request by leorog**

### CODE REVIEW

File: kiik.gemspec [hunk 1]
```diff
@@ -11,6 +11,7 @@ spec = Gem::Specification.new do |s|
   s.license = 'MIT'
 
   s.add_development_dependency('mocha', '~> 0.13')
+  s.add_development_dependency('pry', '~> 0.10.3')
```
Comment (esdrasedu) [inline kiik.gemspec]: remove this

File: Gemfile [hunk 2]
```diff
@@ -2,3 +2,4 @@ source "https://rubygems.org"
 gemspec
 
 gem 'httparty', '>= 0.13.7'
+gem 'pry'
```
Comment (esdrasedu) [inline Gemfile]: NOT BERG

File: lib/kiik/error.rb [hunk 3]
```diff
@@ -1,8 +1,12 @@
 module Kiik
   class KiikError < StandardError
     attr_reader :errors
-    def initialize(body)
-      @errors = [{attr: body['error']['param'], message: body['error']['message']}]
+    def initialize(body = nil)
+      if body
+        @errors = [{attr: body['error']['param'], message: body['error']['message']}] if body
```
Comment (esdrasedu) [inline lib/kiik/error.rb]: this if in the end not sense
and y can do this
```ruby
@errors = if body.nil? ? [] : [{attr: body['error']['param'], message: body['error']['message']}] 
```

File: lib/kiik/rest/create.rb [hunk 4]
```diff
@@ -20,23 +20,36 @@ def create!(params={})
 
         def create(params={})
           result = request(params)
-          raise result if result.instance_of? StandardError or result.instance_of? KiikError
+          raise result if result.instance_of? StandardError
```
Comment (esdrasedu) [inline lib/kiik/rest/create.rb]: do test this, I know this working in JAVA, but in ruby I have doubt

File: lib/kiik/rest/create.rb [hunk 5]
```diff
@@ -20,23 +20,36 @@ def create!(params={})
...
           result
         end
 
         def request(params)
           options = opts.merge!({:body => JSON.generate(params)})
-          response = post url, options
+          response = post(url, options)
+          log_response(response)
```
Comment (esdrasedu) [inline lib/kiik/rest/create.rb]: NOT BERG

File: lib/kiik/rest/create.rb [hunk 6]
```diff
@@ -20,23 +20,36 @@ def create!(params={})
...
+            result.errors << ({attr: 'status', message: 'recused'})
+            build(JSON.parse(response.body), result)
           when 422
             KiikError.new(JSON.parse(response.body))
           else
             StandardError.new(response.message)
           end
         end
+
+        # def log_response(response)
```
Comment (esdrasedu) [inline lib/kiik/rest/create.rb]: NOT BERG

File: lib/kiik/rest/create.rb (hunk 4 above)
Comment (leorog) [inline lib/kiik/rest/create.rb]: worked as designed

**Comment by leorog** [comment]
imgres

File: lib/kiik/rest/create.rb (hunk 4 above)
Comment (esdrasedu) [inline lib/kiik/rest/create.rb]: +1

**Comment by esdrasedu** [comment]
+1

**Comment by jonatandahora** [comment]
+1

File: lib/kiik/error.rb [hunk 7]
```diff
@@ -1,10 +1,10 @@
 module Kiik
   class KiikError < StandardError
     attr_reader :errors
-    def initialize(body)
-      @errors = [{attr: body['error']['param'], message: body['error']['message']}]
+    def initialize(body = nil)
+      @errors = body.nil? ? [] : [{attr: body['error']['param'], message: body['error']['message']}]
     end
-
+
```
Comment (lucca65) [inline lib/kiik/error.rb]: space

**Comment by lucca65** [comment]
+1

**Pull request merged**